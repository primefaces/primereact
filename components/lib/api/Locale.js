import PrimeReact from './PrimeReact';

let locales = {
    en: {
        startsWith: 'Starts with',
        contains: 'Contains',
        notContains: 'Not contains',
        endsWith: 'Ends with',
        equals: 'Equals',
        notEquals: 'Not equals',
        noFilter: 'No Filter',
        filter: 'Filter',
        lt: 'Less than',
        lte: 'Less than or equal to',
        gt: 'Greater than',
        gte: 'Greater than or equal to',
        dateIs: 'Date is',
        dateIsNot: 'Date is not',
        dateBefore: 'Date is before',
        dateAfter: 'Date is after',
        custom: 'Custom',
        clear: 'Clear',
        close: 'Close',
        apply: 'Apply',
        matchAll: 'Match All',
        matchAny: 'Match Any',
        addRule: 'Add Rule',
        removeRule: 'Remove Rule',
        accept: 'Yes',
        reject: 'No',
        choose: 'Choose',
        upload: 'Upload',
        cancel: 'Cancel',
        dayNames: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
        dayNamesShort: ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'],
        dayNamesMin: ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'],
        monthNames: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
        monthNamesShort: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
        today: 'Today',
        weekHeader: 'Wk',
        firstDayOfWeek: 0,
        dateFormat: 'mm/dd/yy',
        weak: 'Weak',
        medium: 'Medium',
        strong: 'Strong',
        passwordPrompt: 'Enter a password',
        emptyFilterMessage: 'No available options',
        emptyMessage: 'No results found',
        aria: {
            trueLabel: 'True',
            falseLabel: 'False',
            nullLabel: 'Not Selected',
            pageLabel: 'Page',
            firstPageLabel: 'First Page',
            lastPageLabel: 'Last Page',
            nextPageLabel: 'Next Page',
            previousPageLabel: 'Previous Page',
            selectLabel: 'Select',
            unselectLabel: 'Unselect',
            expandLabel: 'Expand',
            collapseLabel: 'Collapse'
        }
    }
};

function locale(locale) {
    locale && (PrimeReact.locale = locale);

    return {
        locale: PrimeReact.locale,
        options: locales[PrimeReact.locale]
    };
}

function addLocale(locale, options) {
    locales[locale] = { ...locales['en'], ...options };
}

function updateLocaleOption(key, value, locale) {
    localeOptions(locale)[key] = value;
}

function updateLocaleOptions(options, locale) {
    const _locale = locale || PrimeReact.locale;

    locales[_locale] = { ...locales[_locale], ...options };
}

function localeOption(key, locale) {
    const _locale = locale || PrimeReact.locale;

    try {
        return localeOptions(_locale)[key];
    } catch (error) {
        throw new Error(`The ${key} option is not found in the current locale('${_locale}').`);
    }
}

function ariaLabel(key) {
    const _locale = PrimeReact.locale;

    try {
        return localeOptions(_locale)['aria'][key];
    } catch (error) {
        throw new Error(`The ${key} option is not found in the current locale('${_locale}').`);
    }
}

function localeOptions(locale) {
    const _locale = locale || PrimeReact.locale;

    return locales[_locale];
}

export { locale, addLocale, updateLocaleOption, updateLocaleOptions, localeOption, localeOptions, ariaLabel };
