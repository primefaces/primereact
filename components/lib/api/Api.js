import { FilterMatchMode } from './FilterMatchMode';
import { FilterOperator } from './FilterOperator';
import { FilterService } from './FilterService';
import { addLocale, ariaLabel, locale, localeOption, localeOptions, updateLocaleOption, updateLocaleOptions } from './Locale';
import { MessageSeverity } from './MessageSeverity';
import { PrimeIcons } from './PrimeIcons';
import PrimeReact from './PrimeReact';
import { SortOrder } from './SortOrder';

export default PrimeReact;

export { locale, addLocale, ariaLabel, localeOption, localeOptions, updateLocaleOption, updateLocaleOptions, PrimeIcons, MessageSeverity, FilterMatchMode, FilterOperator, FilterService, SortOrder };
