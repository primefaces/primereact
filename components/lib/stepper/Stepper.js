import React from 'react';
import { PrimeReactContext } from '../api/Api';
import { useHandleStyle } from '../componentbase/ComponentBase';
import { CSSTransition } from '../csstransition/CSSTransition';
import { useMergeProps, useMountEffect } from '../hooks/Hooks';
import { UniqueComponentId, classNames } from '../utils/Utils';
import { StepperBase } from './StepperBase';
import { StepperContent } from './StepperContent';
import { StepperHeader } from './StepperHeader';
import { StepperSeparator } from './StepperSeparator';

export const Stepper = React.memo(
    React.forwardRef((inProps, ref) => {
        const mergeProps = useMergeProps();
        const context = React.useContext(PrimeReactContext);
        const props = StepperBase.getProps(inProps, context);
        const { ptm, cx, isUnstyled, ptmo } = StepperBase.setMetaData({
            props
        });
        const [idState, setIdState] = React.useState(props.id);
        const [activeStepState, setActiveStepState] = React.useState(props.activeStep);
        const navRef = React.useRef();

        useHandleStyle(StepperBase.css.styles, isUnstyled, { name: 'stepper' });

        useMountEffect(() => {
            if (!idState) {
                setIdState(UniqueComponentId());
            }
        });

        const getStepProp = (step, name) => {
            return step?.props?.[name];
        };

        const getStepKey = (step, index) => {
            return getStepProp(step, 'header') || index;
        };

        const isStep = (child) => {
            return child.type.displayName === 'StepperPanel';
        };

        const isStepActive = (index) => {
            return activeStepState === index;
        };

        const isItemDisabled = (index) => {
            return props.linear && !isStepActive(index);
        };

        const updateActiveStep = (event, index) => {
            setActiveStepState(index);

            props.onChangeStep && props.onChangeStep({ originalEvent: event, index });
        };

        const getStepHeaderActionId = (index) => {
            return `${idState}_${index}_header_action`;
        };

        const getStepContentId = (index) => {
            return `${idState}_${index}content`;
        };

        const stepperPanels = () => {
            return React.Children.toArray(props.children).reduce((stepperpanels, child) => {
                if (isStep(child)) {
                    stepperpanels.push(child);
                } else if (child && Array.isArray(child)) {
                    React.Children.toArray(child.props.children).forEach((nestedChild) => {
                        if (isStep(nestedChild)) {
                            stepperpanels.push(nestedChild);
                        }
                    });
                }

                return stepperpanels;
            }, []);
        };

        const prevCallback = (event, index) => {
            if (index !== 0) {
                updateActiveStep(event, index - 1);
            }
        };

        const nextCallback = (event, index) => {
            if (index !== stepperPanels().length - 1) {
                updateActiveStep(event, index + 1);
            }
        };

        const getStepPT = (step, key, index) => {
            const count = stepperPanels().length;

            const stepMetaData = {
                props: step.props,
                parent: {
                    props: props
                },
                context: {
                    index,
                    count,
                    first: index === 0,
                    last: index === count - 1,
                    active: isStepActive(index),
                    highlighted: index < activeStepState,
                    disabled: isItemDisabled(index)
                }
            };

            return mergeProps(ptm(`stepperpanel.${key}`, { stepperpanel: stepMetaData }), ptm(`stepperpanel.${key}`, stepMetaData), ptmo(getStepProp(step, 'pt'), key, stepMetaData));
        };

        const onItemClick = (event, index) => {
            if (props.linear) {
                event.preventDefault();

                return;
            }

            if (index !== activeStepState) {
                updateActiveStep(event, index);
            }
        };

        const createPanel = () => {
            return stepperPanels().map((step, index) => {
                const panelProps = mergeProps(
                    {
                        className: classNames(cx('stepper.header', { isStepActive, isItemDisabled, step, index })),
                        'aria-current': isStepActive(index) && 'step',
                        role: 'presentation',
                        'data-p-highlight': isStepActive(index),
                        'data-p-disabled': isItemDisabled(index),
                        'data-p-active': isStepActive(index)
                    },
                    ptm('stepperpanel')
                );

                return (
                    <li key={getStepKey(step, index)} {...panelProps}>
                        <StepperHeader
                            id={getStepHeaderActionId(index)}
                            template={step.children?.header}
                            stepperpanel={step}
                            index={index}
                            disabled={isItemDisabled(index)}
                            active={isStepActive(index)}
                            highlighted={index < activeStepState}
                            ariaControls={getStepContentId(index)}
                            clickCallback={onItemClick}
                            getStepPT={getStepPT}
                            getStepProp={getStepProp}
                            cx={cx}
                        />
                        {index !== stepperPanels().length - 1 && (
                            <StepperSeparator template={step.children?.separator} separatorClass={cx('stepper.separator')} stepperpanel={step} index={index} active={isStepActive(index)} highlighted={index < activeStepState} getStepPT={getStepPT} />
                        )}
                    </li>
                );
            });
        };

        React.useImperativeHandle(ref, () => ({
            getElement: () => navRef.current,
            getActiveStep: () => activeStepState,
            setActiveStep: (step) => setActiveStepState(step),
            nextCallback: (e) => nextCallback(e, activeStepState),
            prevCallback: (e) => prevCallback(e, activeStepState)
        }));

        const createPanelContent = () => {
            return stepperPanels().map((step, index) => {
                if (!isStepActive(index)) {
                    return null;
                }

                return (
                    <StepperContent
                        key={getStepContentId(index)}
                        id={getStepContentId(index)}
                        tempate={step?.children?.content}
                        stepperpanel={step}
                        index={index}
                        active={isStepActive(index)}
                        highlighted={index < activeStepState}
                        clickCallback={onItemClick}
                        prevCallback={prevCallback}
                        nextCallback={nextCallback}
                        getStepPT={getStepPT}
                        ariaLabelledby={getStepHeaderActionId(index)}
                        ptm={ptm}
                        cx={cx}
                    />
                );
            });
        };

        const createHorizontal = () => {
            const items = createPanel();
            const navProps = mergeProps(
                {
                    className: classNames(cx('nav')),
                    ref: navRef
                },
                ptm('nav')
            );

            const panelContainerProps = mergeProps(
                {
                    className: cx('panelContainer')
                },
                ptm('panelContainer')
            );

            return (
                <>
                    <ul {...navProps}>{items}</ul>
                    <div {...panelContainerProps}>{createPanelContent()}</div>
                </>
            );
        };

        const createVertical = () => {
            return stepperPanels().map((step, index) => {
                const contentRef = React.createRef(null);

                const navProps = mergeProps(
                    {
                        ref: navRef,
                        className: cx('panel', { props, index, isStepActive }),
                        'aria-current': isStepActive(index) && 'step',
                        ...getStepPT(step, 'root', index),
                        ...getStepPT(step, 'panel', index),
                        'data-p-highlight': isStepActive(index),
                        'data-p-disabled': isItemDisabled(index),
                        'data-p-active': isStepActive(index)
                    },
                    ptm('nav')
                );

                const headerProps = mergeProps({
                    className: cx('stepper.header', { step, isStepActive, isItemDisabled, index }),
                    ...getStepPT(step, 'header', index)
                });

                const transitionProps = mergeProps({
                    classNames: cx('stepper.content'),
                    ...getStepPT(step, 'transition', index),
                    timeout: { enter: 1000, exit: 450 },
                    in: isStepActive(index),
                    unmountOnExit: true
                });

                const toggleableContentProps = mergeProps({
                    ref: contentRef,
                    className: cx('stepper.toggleableContent'),
                    ...getStepPT(step, 'toggleableContent', index)
                });

                return (
                    <div key={getStepKey(step, index)} {...navProps}>
                        <div {...headerProps}>
                            <StepperHeader
                                id={getStepHeaderActionId(index)}
                                template={step.children?.header}
                                stepperpanel={step}
                                index={index}
                                disabled={isItemDisabled(index)}
                                active={isStepActive(index)}
                                highlighted={index < activeStepState}
                                ariaControls={getStepContentId(index)}
                                clickCallback={onItemClick}
                                getStepPT={getStepPT}
                                getStepProp={getStepProp}
                                cx={cx}
                            />
                        </div>
                        <CSSTransition nodeRef={contentRef} {...transitionProps}>
                            <div {...toggleableContentProps}>
                                {index !== stepperPanels().length - 1 && (
                                    <StepperSeparator
                                        template={step.children?.separator}
                                        separatorClass={cx('stepper.separator')}
                                        stepperpanel={step}
                                        index={index}
                                        active={isStepActive(index)}
                                        highlighted={index < activeStepState}
                                        getStepPT={getStepPT}
                                    />
                                )}
                                <StepperContent
                                    key={getStepContentId(index)}
                                    id={getStepContentId(index)}
                                    tempate={step?.children?.content}
                                    stepperpanel={step}
                                    index={index}
                                    active={isStepActive(index)}
                                    highlighted={index < activeStepState}
                                    clickCallback={onItemClick}
                                    prevCallback={prevCallback}
                                    nextCallback={nextCallback}
                                    getStepPT={getStepPT}
                                    ariaLabelledby={getStepHeaderActionId(index)}
                                    ptm={ptm}
                                    cx={cx}
                                />
                            </div>
                        </CSSTransition>
                    </div>
                );
            });
        };

        const rootProps = mergeProps(
            {
                className: classNames(cx('root')),
                role: 'tablist'
            },
            StepperBase.getOtherProps(props),
            ptm('root')
        );

        return (
            <div {...rootProps}>
                {props.start && props.start()}
                {props.orientation === 'horizontal' && createHorizontal()}
                {props.orientation === 'vertical' && createVertical()}
                {props.end && props.end()}
            </div>
        );
    })
);

StepperBase.displayName = 'StepperBase';
