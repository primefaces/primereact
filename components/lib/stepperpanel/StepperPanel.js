import * as React from 'react';
import { PrimeReactContext } from '../api/Api';
import { useHandleStyle } from '../componentbase/ComponentBase';
import { StepperPanelBase } from './StepperPanelBase';

export const StepperPanel = React.memo(
    React.forwardRef((inProps, ref) => {
        const context = React.useContext(PrimeReactContext);
        const props = StepperPanelBase.getProps(inProps, context);
        const { isUnstyled } = StepperPanelBase.setMetaData({
            props
        });

        useHandleStyle(StepperPanelBase.css.styles, isUnstyled, { name: 'StepperPanel' });

        return <span ref={ref}>{props.children}</span>;
    })
);

StepperPanel.displayName = 'StepperPanel';
