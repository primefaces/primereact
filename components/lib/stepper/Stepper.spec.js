import '@testing-library/jest-dom';
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { snapshot } from '../../test';
import { PrimeReactProvider } from '../api/Api';
import { Stepper } from './Stepper';
import { StepperPanel } from '../stepperpanel/StepperPanel';

function assertSelectedStep(expectedIndex) {
    const selectedStep = screen.getByRole('presentation', { current: 'step' });
    const expectedStep = screen.getByRole('list').children.item(expectedIndex);

    expect(expectedStep).toEqual(selectedStep);
}

describe('Stepper', () => {
    snapshot(
        <PrimeReactProvider>
            <Stepper />
        </PrimeReactProvider>,
        'default'
    );
    snapshot(
        <Stepper>
            <StepperPanel header="Header I"></StepperPanel>
            <StepperPanel header="Header II"></StepperPanel>
            <StepperPanel header="Header III"></StepperPanel>
        </Stepper>,
        'panels'
    );
    snapshot(
        <Stepper orientation="vertical">
            <StepperPanel header="Header I"></StepperPanel>
            <StepperPanel header="Header II"></StepperPanel>
            <StepperPanel header="Header III"></StepperPanel>
        </Stepper>,
        'vertical'
    );
    snapshot(
        <Stepper headerPosition="top">
            <StepperPanel header="Header I"></StepperPanel>
            <StepperPanel header="Header II"></StepperPanel>
        </Stepper>,
        'header position top'
    );
    snapshot(
        <Stepper headerPosition="right">
            <StepperPanel header="Header I"></StepperPanel>
            <StepperPanel header="Header II"></StepperPanel>
        </Stepper>,
        'header position right'
    );
    snapshot(
        <Stepper headerPosition="bottom">
            <StepperPanel header="Header I"></StepperPanel>
            <StepperPanel header="Header II"></StepperPanel>
        </Stepper>,
        'header position bottom'
    );
    snapshot(
        <Stepper headerPosition="left">
            <StepperPanel header="Header I"></StepperPanel>
            <StepperPanel header="Header II"></StepperPanel>
        </Stepper>,
        'header position left'
    );

    test('Step should have aria step when selected', async () => {
        render(
            <Stepper>
                <StepperPanel header="Header I"></StepperPanel>
                <StepperPanel header="Header II"></StepperPanel>
                <StepperPanel header="Header III"></StepperPanel>
            </Stepper>
        );

        await userEvent.click(screen.getByRole('tab', { name: '3 Header III' }));
        assertSelectedStep(2);
    });

    test('Control active step from outside', async () => {
        render(
            <Stepper activeStep={3}>
                <StepperPanel header="Header I"></StepperPanel>
                <StepperPanel header="Header II"></StepperPanel>
                <StepperPanel header="Header III"></StepperPanel>
                <StepperPanel header="Header IV"></StepperPanel>
                <StepperPanel header="Header VI"></StepperPanel>
            </Stepper>
        );

        assertSelectedStep(3);
    });

    test('Changing step should trigger onChangeStep callback', async () => {
        const onChangeStep = jest.fn();

        render(
            <Stepper onChangeStep={onChangeStep}>
                <StepperPanel header="Header I"></StepperPanel>
                <StepperPanel header="Header II"></StepperPanel>
                <StepperPanel header="Header III"></StepperPanel>
            </Stepper>
        );

        await userEvent.click(screen.getByRole('tab', { name: '2 Header II' }));
        expect(onChangeStep).toHaveBeenCalledTimes(1);

        await userEvent.click(screen.getByRole('tab', { name: '3 Header III' }));
        expect(onChangeStep).toHaveBeenCalledTimes(2);
    });

    test('Change steps with buttons', async () => {
        const stepperRef = React.createRef(null);
        const onChangeStep = jest.fn();

        render(
            <div>
                <Stepper ref={stepperRef} onChangeStep={onChangeStep}>
                    <StepperPanel header="Header I"></StepperPanel>
                    <StepperPanel header="Header II"></StepperPanel>
                    <StepperPanel header="Header III"></StepperPanel>
                </Stepper>
                <button onClick={() => stepperRef.current.nextCallback()}>Next</button>
                <button onClick={() => stepperRef.current.previousCallback()}>Previous</button>
            </div>
        );

        const nextButton = screen.getByRole('button', { name: 'Next' });

        await userEvent.click(nextButton);

        expect(onChangeStep).toHaveBeenCalledTimes(1);

        assertSelectedStep(1);
    });
});
