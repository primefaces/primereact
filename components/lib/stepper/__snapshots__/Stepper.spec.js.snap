// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper default 1`] = `
<div>
  <div
    class="p-stepper p-component p-stepper-horizontal"
    data-pc-name="stepper"
    data-pc-section="root"
    role="tablist"
  >
    <ul
      class="p-stepper-nav"
      data-pc-section="nav"
    />
    <div
      class="p-stepper-panels"
      data-pc-section="panelcontainer"
    />
  </div>
</div>
`;

exports[`Stepper header position bottom 1`] = `
<div>
  <div
    class="p-stepper p-component p-stepper-horizontal"
    data-pc-name="stepper"
    data-pc-section="root"
    role="tablist"
  >
    <ul
      class="p-stepper-nav"
      data-pc-section="nav"
    >
      <li
        aria-current="step"
        class="p-stepper-header p-highlight p-stepper-header-bottom"
        data-p-active="true"
        data-p-disabled="false"
        data-p-highlight="true"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_6_0content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_6_0_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            1
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header I
          </span>
        </button>
        <span
          aria-hidden="true"
          class="p-stepper-separator"
          data-pc-section="separator"
        />
      </li>
      <li
        aria-current="false"
        class="p-stepper-header p-stepper-header-bottom"
        data-p-active="false"
        data-p-disabled="false"
        data-p-highlight="false"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_6_1content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_6_1_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            2
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header II
          </span>
        </button>
      </li>
    </ul>
    <div
      class="p-stepper-panels"
      data-pc-section="panelcontainer"
    >
      <div
        aria-labelledby="pr_id_6_0_header_action"
        class="p-stepper-content"
        data-p-active="true"
        data-pc-name=""
        data-pc-section="content"
        id="pr_id_6_0content"
        role="tabpanel"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper header position left 1`] = `
<div>
  <div
    class="p-stepper p-component p-stepper-horizontal"
    data-pc-name="stepper"
    data-pc-section="root"
    role="tablist"
  >
    <ul
      class="p-stepper-nav"
      data-pc-section="nav"
    >
      <li
        aria-current="step"
        class="p-stepper-header p-highlight p-stepper-header-left"
        data-p-active="true"
        data-p-disabled="false"
        data-p-highlight="true"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_7_0content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_7_0_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            1
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header I
          </span>
        </button>
        <span
          aria-hidden="true"
          class="p-stepper-separator"
          data-pc-section="separator"
        />
      </li>
      <li
        aria-current="false"
        class="p-stepper-header p-stepper-header-left"
        data-p-active="false"
        data-p-disabled="false"
        data-p-highlight="false"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_7_1content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_7_1_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            2
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header II
          </span>
        </button>
      </li>
    </ul>
    <div
      class="p-stepper-panels"
      data-pc-section="panelcontainer"
    >
      <div
        aria-labelledby="pr_id_7_0_header_action"
        class="p-stepper-content"
        data-p-active="true"
        data-pc-name=""
        data-pc-section="content"
        id="pr_id_7_0content"
        role="tabpanel"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper header position right 1`] = `
<div>
  <div
    class="p-stepper p-component p-stepper-horizontal"
    data-pc-name="stepper"
    data-pc-section="root"
    role="tablist"
  >
    <ul
      class="p-stepper-nav"
      data-pc-section="nav"
    >
      <li
        aria-current="step"
        class="p-stepper-header p-highlight p-stepper-header-right"
        data-p-active="true"
        data-p-disabled="false"
        data-p-highlight="true"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_5_0content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_5_0_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            1
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header I
          </span>
        </button>
        <span
          aria-hidden="true"
          class="p-stepper-separator"
          data-pc-section="separator"
        />
      </li>
      <li
        aria-current="false"
        class="p-stepper-header p-stepper-header-right"
        data-p-active="false"
        data-p-disabled="false"
        data-p-highlight="false"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_5_1content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_5_1_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            2
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header II
          </span>
        </button>
      </li>
    </ul>
    <div
      class="p-stepper-panels"
      data-pc-section="panelcontainer"
    >
      <div
        aria-labelledby="pr_id_5_0_header_action"
        class="p-stepper-content"
        data-p-active="true"
        data-pc-name=""
        data-pc-section="content"
        id="pr_id_5_0content"
        role="tabpanel"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper header position top 1`] = `
<div>
  <div
    class="p-stepper p-component p-stepper-horizontal"
    data-pc-name="stepper"
    data-pc-section="root"
    role="tablist"
  >
    <ul
      class="p-stepper-nav"
      data-pc-section="nav"
    >
      <li
        aria-current="step"
        class="p-stepper-header p-highlight p-stepper-header-top"
        data-p-active="true"
        data-p-disabled="false"
        data-p-highlight="true"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_4_0content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_4_0_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            1
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header I
          </span>
        </button>
        <span
          aria-hidden="true"
          class="p-stepper-separator"
          data-pc-section="separator"
        />
      </li>
      <li
        aria-current="false"
        class="p-stepper-header p-stepper-header-top"
        data-p-active="false"
        data-p-disabled="false"
        data-p-highlight="false"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_4_1content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_4_1_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            2
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header II
          </span>
        </button>
      </li>
    </ul>
    <div
      class="p-stepper-panels"
      data-pc-section="panelcontainer"
    >
      <div
        aria-labelledby="pr_id_4_0_header_action"
        class="p-stepper-content"
        data-p-active="true"
        data-pc-name=""
        data-pc-section="content"
        id="pr_id_4_0content"
        role="tabpanel"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper panels 1`] = `
<div>
  <div
    class="p-stepper p-component p-stepper-horizontal"
    data-pc-name="stepper"
    data-pc-section="root"
    role="tablist"
  >
    <ul
      class="p-stepper-nav"
      data-pc-section="nav"
    >
      <li
        aria-current="step"
        class="p-stepper-header p-highlight p-stepper-header-right"
        data-p-active="true"
        data-p-disabled="false"
        data-p-highlight="true"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_2_0content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_2_0_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            1
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header I
          </span>
        </button>
        <span
          aria-hidden="true"
          class="p-stepper-separator"
          data-pc-section="separator"
        />
      </li>
      <li
        aria-current="false"
        class="p-stepper-header p-stepper-header-right"
        data-p-active="false"
        data-p-disabled="false"
        data-p-highlight="false"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_2_1content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_2_1_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            2
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header II
          </span>
        </button>
        <span
          aria-hidden="true"
          class="p-stepper-separator"
          data-pc-section="separator"
        />
      </li>
      <li
        aria-current="false"
        class="p-stepper-header p-stepper-header-right"
        data-p-active="false"
        data-p-disabled="false"
        data-p-highlight="false"
        data-pc-section="header"
        role="presentation"
      >
        <button
          aria-controls="pr_id_2_2content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_2_2_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            3
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header III
          </span>
        </button>
      </li>
    </ul>
    <div
      class="p-stepper-panels"
      data-pc-section="panelcontainer"
    >
      <div
        aria-labelledby="pr_id_2_0_header_action"
        class="p-stepper-content"
        data-p-active="true"
        data-pc-name=""
        data-pc-section="content"
        id="pr_id_2_0content"
        role="tabpanel"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper vertical 1`] = `
<div>
  <div
    class="p-stepper p-component p-stepper-vertical"
    data-pc-name="stepper"
    data-pc-section="root"
    role="tablist"
  >
    <div
      aria-current="step"
      class="p-stepper-panel p-stepper-panel-active"
      data-p-active="true"
      data-p-disabled="false"
      data-p-highlight="true"
      data-pc-name=""
      data-pc-section="panel"
    >
      <div
        class="p-stepper-header p-highlight"
        data-pc-section="header"
      >
        <button
          aria-controls="pr_id_3_0content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_3_0_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            1
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header I
          </span>
        </button>
      </div>
      <div
        class="p-stepper-toggleable-content"
        data-pc-section="toggleablecontent"
      >
        <span
          aria-hidden="true"
          class="p-stepper-separator"
          data-pc-section="separator"
        />
        <div
          aria-labelledby="pr_id_3_0_header_action"
          class="p-stepper-content p-toggleable-content"
          data-p-active="true"
          data-pc-name=""
          data-pc-section="content"
          id="pr_id_3_0content"
          role="tabpanel"
        >
          <span />
        </div>
      </div>
    </div>
    <div
      aria-current="false"
      class="p-stepper-panel"
      data-p-active="false"
      data-p-disabled="false"
      data-p-highlight="false"
      data-pc-name=""
      data-pc-section="panel"
    >
      <div
        class="p-stepper-header"
        data-pc-section="header"
      >
        <button
          aria-controls="pr_id_3_1content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_3_1_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            2
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header II
          </span>
        </button>
      </div>
    </div>
    <div
      aria-current="false"
      class="p-stepper-panel"
      data-p-active="false"
      data-p-disabled="false"
      data-p-highlight="false"
      data-pc-name=""
      data-pc-section="panel"
    >
      <div
        class="p-stepper-header"
        data-pc-section="header"
      >
        <button
          aria-controls="pr_id_3_2content"
          class="p-stepper-action p-component"
          data-pc-section="action"
          id="pr_id_3_2_header_action"
          role="tab"
          type="button"
        >
          <span
            class="p-stepper-number"
            data-pc-section="number"
          >
            3
          </span>
          <span
            class="p-stepper-title"
            data-pc-section="title"
          >
            Header III
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
